### DISCUSSION TITLE
Just a few improvement suggestions

### DISCUSSION THREAD

**Discussion by danroth77** [tolwi/hassio-ecoflow-cloud#161] [General]
This integration works quite well, except for 2 little details:

1. PowerStream does not show up on the energy dashboard - this means I cannot integrate Solar 1 & Solar 2
2. In automations, PowerStream shows up, but there are no Actions to be selected, plus no sensors available

I assume these are 2 small fixes. I appreciate you help in getting this to work.

**Comment by Freybeuter** [comment]
I'd love to be able to set my household power demand with an automation.

**Reply by Fyr42** [reply]
It's done with these "SmartPlugs" that talk to the powerstream

**Comment by danroth77** [comment]
I would love that too! Did anyone get the Powerstream integration to work? I don’t get any values coming in. Delta Pro works like a charm!

Am 21.10.2023 um 16:35 schrieb Freybeuter ***@***.***>:

﻿

I'd love to be able to set my household power demand with an automation.

**Comment by Fyr42** [comment]
You have to build an helper for each panel. PowerStream gives only instant power (watt)  while Energy Dashboard uses cumulative power (watt per hour)

See https://www.home-assistant.io/integrations/integration/ to build kWh from kW

**Comment by danroth77** [comment]
The SmartPlugs already do their job, yes. They take whatever power is available (solar and or battery) and feed the devices attached to them. I’m referring to being able to set the “base” power (what the rest of the appliances, that are not hooked up on a SmartPlug are consuming). I’m not sure if “base” power is the correct term, I use the German version of the Ecoflow app. Mine is set to 150 Watts which works pretty well. There are cases, where it would be helpful to set it to another value.
Very cool, would be the ability to switch the Powerstream from “battery mode” (save incoming solar to battery) to “input to house and Smartplugs” (not sure of the terms, as I’m using the German version). The app only allows for simple time based automation, not rule based like HA.

Am 21.10.2023 um 16:49 schrieb Fyr42 ***@***.***>:

﻿

It's done with these "SmartPlugs" that talk to the powerstream

**Reply by Freybeuter** [reply]
Exactly. I got appliances not connected to a smart plug and it would be great to be able to automate the whole process. If the sun is up and the estimated energy now is above 100 watts? Switch to battery prio and set household demand to 600 watts. Undo the changes once the estimated energy production or the actual energy production goes below 100 watts.  I guess, I could put a smart plug on every appliance, but they are sort of expensive and they use some power themselves. 
We already have the ability to switch the power supply mode between battery first or house first, but no way to change the household power demand.

**Comment by danroth77** [comment]
Cool! Thanks for the tip! How do I get Powerstream to give me the values, as mine does not give me any values. Delta Pro works like a charm!

Am 21.10.2023 um 16:55 schrieb Fyr42 ***@***.***>:

﻿

You have to build an helper for each panel. PowerStream gives only instant power (watt) while Energy Dashboard uses cumulative power (watt per hour)

See https://www.home-assistant.io/integrations/integration/ to build kWh from kW

**Comment by danroth77** [comment]
Would you share how to implement the switching of the power supply mode in HA from battery first to house first? The control doesn’t show up in HA for me.

Am 21.10.2023 um 22:30 schrieb Freybeuter ***@***.***>:

﻿

Exactly. I got appliances not connected to a smart plug and it would be great to be able to automate the whole process. If the sun is up and the estimated energy now is above 100 watts? Switch to battery prio and set household demand to 600 watts. Undo the changes once the estimated energy production or the actual energy production goes below 100 watts. I guess, I could put a smart plug on every appliance, but they are sort of expensive and they use some power themselves.
We already have the ability to switch the power supply mode between battery first or house first, but no way to change the household power demand.

**Reply by Freybeuter** [reply]
I did the usual thing after installing and configuring the custom Integration: Start editing the dashboard I want to use for the battery, press the button on the lower right corner "add card", this opens a new window where I sort by entity and the name of the entity is "Battery Prio Solar Charging" or "switch.battery_prio_solar_charging". Click the check box "𐄂" to the left of the entity, press "continue" on the lower right and "add to dashboard".

**Comment by danroth77** [comment]
Thanks for the information. I don’t have the entity Battery prio solar charging. Is there a trick to add it as an entity? I’ve tried adding the integration again and reloading the yaml but it doesn’t show up.

Am 22.10.2023 um 15:35 schrieb Freybeuter ***@***.***>:

﻿

I did the usual thing after installing and configuring the custom Integration: Start editing the dashboard I want to use for the battery, press the button on the lower right corner "add card", this opens a new window where I sort by entity and the name of the entity is "Battery Prio Solar Charging" or "switch.battery_prio_solar_charging". Click the check box "𐄂" to the left of the entity, press "continue" on the lower right and "add to dashboard".
